src/java.base/share/classes/java/util/Calendar.java

```diff
@@ -1490,26 +1490,21 @@ public Calendar build() {
                 }
             }
             switch (type) {
-            case "gregory":
-                cal = new GregorianCalendar(zone, locale, true);
-                break;
-            case "iso8601":
-                GregorianCalendar gcal = new GregorianCalendar(zone, locale, true);
-                // make gcal a proleptic Gregorian
-                gcal.setGregorianChange(new Date(Long.MIN_VALUE));
-                // and week definition to be compatible with ISO 8601
-                setWeekDefinition(MONDAY, 4);
-                cal = gcal;
-                break;
-            case "buddhist":
-                cal = new BuddhistCalendar(zone, locale);
-                cal.clear();
-                break;
-            case "japanese":
-                cal = new JapaneseImperialCalendar(zone, locale, true);
-                break;
-            default:
-                throw new IllegalArgumentException("unknown calendar type: " + type);
+                case "gregory" -> cal = new GregorianCalendar(zone, locale, true);
+                case "iso8601" -> {
+                    GregorianCalendar gcal = new GregorianCalendar(zone, locale, true);
+                    // make gcal a proleptic Gregorian
+                    gcal.setGregorianChange(new Date(Long.MIN_VALUE));
+                    // and week definition to be compatible with ISO 8601
+                    setWeekDefinition(MONDAY, 4);
+                    cal = gcal;
+                }
+                case "buddhist" -> {
+                    cal = new BuddhistCalendar(zone, locale);
+                    cal.clear();
+                }
+                case "japanese" -> cal = new JapaneseImperialCalendar(zone, locale, true);
+                default -> throw new IllegalArgumentException("unknown calendar type: " + type);
```

Comment:
In this case, it would be good to yield the result of the switch expression and assign that to a local, rather than assigning in each of the case arms, e.g:

    final Calendar cal = switch (type) {
        case "gregory" -> new GregorianCalendar(zone, locale, true);
        case "iso8601" -> {
            GregorianCalendar gcal = new GregorianCalendar(zone, locale, true);
            // make gcal a proleptic Gregorian
            gcal.setGregorianChange(new Date(Long.MIN_VALUE));
            // and week definition to be compatible with ISO 8601
            setWeekDefinition(MONDAY, 4);
            yield gcal;
        }
        case "buddhist" -> {
            var buddhistCalendar = new BuddhistCalendar(zone, locale);
            buddhistCalendar.clear();
            yield buddhistCalendar;
        }
        case "japanese" -> new JapaneseImperialCalendar(zone, locale, true);
        default -> throw new IllegalArgumentException("unknown calendar type: " + type);
    };

Reply:
Done, thanks!